src/projects/projects.service.ts

```diff
@@ -14,33 +14,84 @@ export class ProjectsService {
     return exclude(project, ['id', 'userId']);
   }
 
-  async create(user: User, project: CreateProjectDto): Promise<SanitizedProject> {
-    return this.sanitizeOutput(await this.prisma.project.create({
+  async create(userId: number, project: CreateProjectDto): Promise<SanitizedProject> {
+    const createdProject = await this.prisma.project.create({
       data: {
         name: project.name,
         uuid: uuidv4(),
-        userId: user.id,
-      }
-    }));
+        userId: userId,
+      },
+    });
+
+    const sanitizedOutput = this.sanitizeOutput(createdProject);
+
+    return {
+      ...sanitizedOutput,
+      uuid: createdProject.uuid,
+    };
   }
 
-  async findOne(uuid: string): Promise<SanitizedProject|null> {
-    const project = await this.prisma.project.findUnique({
-      // FIXME: ensure one can only find their projects and not somebody else's
-      where: {uuid},
+  async findOne(uuid: string, userId: number): Promise<SanitizedProject | null> {
+    const user = await this.prisma.user.findUnique({
+      where: { id: userId },
     });
 
-    if (project) {
-      return this.sanitizeOutput(project);
-    } else {
-      return null;
+    // if user is admin, we can check
+    if (user.isAdmin) {
+      return (
+        (await this.prisma.project.findFirst({
+          where: { uuid },
+        })) ?? null
+      );
     }
+
+    return (
+      (await this.prisma.project.findFirst({
+        where: { userId, uuid },
+      })) ?? null
+    );
+  }
+
+  async findOneByName(name: string, userId: number): Promise<SanitizedProject | null> {
+    const project = await this.prisma.project.findFirst({
+      where: { userId, name: name },
+    });
+    return project ?? null;
   }
 
   async delete(uuid: string) {
-    // FIXME: ensure the requesteer has the rights to delete the project
+    // Retrieve the project before deleting it
+    const project = await this.prisma.project.findFirst({
+      where: { uuid },
+    });
+    if (!project) {
+      throw new NotFoundException('Project not found');
+    }
+
+    // Delete the project
     await this.prisma.project.delete({
-      where: {uuid}
+      where: { uuid },
     });
+
+    return project;
+  }
+
+  async findAll(userId: number): Promise<SanitizedProject[]> {
+    // check if user is admin
+    const user = await this.prisma.user.findUnique({
+      where: { id: userId },
+    });
+
+    // If the user is admin we return every project in the database
+    if (user.isAdmin) {
+      const projects = await this.prisma.project.findMany();
+      return projects.map((project) => project);
```

Comment:
`sanitizeOutput()` or not `map(), that is the question

(I prefer with sanitize, to have the same data structure on the front)

Reply:
``` return projects.map((project) => this.sanitizeOutput(project));```
done :+1: